packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -172,6 +177,96 @@ export const transformPlotlyJsonToGVBCProps = (
   };
 };
 
+// TODO: Add support for padding in continuous axis
+// FIXME: Vertical bar chart displays duplicate legend buttons
+export const transformPlotlyJsonToVBCProps = (
+  jsonObj: any,
+  colorMap: React.MutableRefObject<Map<string, string>>,
+): IVerticalBarChartProps => {
+  const { data, layout } = jsonObj;
+  const vbcData: IVerticalBarChartDataPoint[] = [];
+
+  data.forEach((series: any, index: number) => {
+    if (!series.x) {
+      return;
+    }
+
+    const scale = d3ScaleLinear()
+      .domain(d3Extent<number>(series.x) as [number, number])
+      .nice();
+    let [xMin, xMax] = scale.domain();
+
+    xMin = typeof series.xbins?.start === 'number' ? series.xbins.start : xMin;
+    xMax = typeof series.xbins?.end === 'number' ? series.xbins.end : xMax;
+
+    const bin = d3Bin().domain([xMin, xMax]);
+
+    if (typeof series.xbins?.size === 'number') {
+      const thresholds: number[] = [];
+      let th = xMin;
+      const precision = d3PrecisionFixed(series.xbins.size);
+      const format = d3Format(`.${precision}f`);
+
+      while (th < xMax + series.xbins.size) {
+        thresholds.push(parseFloat(format(th)));
+        th += series.xbins.size;
+      }
+
+      xMin = thresholds[0];
+      xMax = thresholds[thresholds.length - 1];
+      bin.domain([xMin, xMax]).thresholds(thresholds);
+    }
+
+    const buckets = bin(series.x);
+    const totalDataPoints = d3Merge(buckets).length;
+
+    buckets.forEach(bucket => {
+      if (bucket.length < 1) {
+        return;
+      }
```

Comment:
are you removing bucket with length = 0

Reply:
I found some cases where the last bin had the same x0 and x1. So, I applied this check to prevent calculation issues below. But now, I think it would be better to replace it with `x0 >= x1`. What do you suggest?

Reply:
Lets see how plotly is handling this case. We can debug their code if needed.
But a bin could have y = 0. In either case we dont render a bar. But let's get the computations complete.

Reply:
removed this outer check and added inner checks while calculating y